### PR TITLE
Spark: Qualify destination location before check when snapshot

### PR DESCRIPTION
**Pull request by uncleGen**
add new UT

### CODE REVIEW

File: spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java [hunk 1]
```diff
@@ -135,10 +140,23 @@ protected TableCatalog checkSourceCatalog(CatalogPlugin catalog) {
 
   @Override
   public SnapshotAction withLocation(String location) {
-    Preconditions.checkArgument(!sourceTableLocation().equals(location),
```
Comment (uncleGen) [inline spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:138]: `sourceTableLocation` is converted from URI, while `location` comes from SQL command. We should qualify the `location` before check equals.

File: spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java [hunk 2]
```diff
@@ -135,10 +140,23 @@ protected TableCatalog checkSourceCatalog(CatalogPlugin catalog) {
...
+
+  private String makeQualifiedLocation(String location) {
+    try {
+      Path path = new Path(location);
+      Configuration conf = spark().sessionState().newHadoopConf();
+      FileSystem fs = FileSystem.get(path.toUri(), conf);
+      return CatalogUtils.URIToString(fs.makeQualified(path).toUri());
+    } catch (IOException ioe) {
+      LOG.error("Failed to initialize hadoop filesystem.", ioe);
+      return location;
```
Comment (uncleGen) [inline spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:159]: If failed, back to location. Any suggestions?

**Comment by uncleGen** [comment]
cc @RussellSpitzer

File: spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java (hunk 2 above)
Comment (RussellSpitzer) [reply spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:159]: We should probably just fail, after all if the location string cannot be resolved isn't it going to fail anyway?

File: spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java (hunk 1 above)
Comment (RussellSpitzer) [reply spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:138]: Good catch!

Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:138]: We should probably also check if two locations overlap. We will corrupt the original table if the source location is within the original location.

Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:138]: Does it mean we need to normalize and resolve the locations?

File: spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java (hunk 2 above)
Comment (uncleGen) [reply spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java:159]: make sense @RussellSpitzer

**Pull request closed without merging**